ReactiveCocoa/Swift/Signal.swift

```diff
@@ -262,6 +262,30 @@ public enum FlattenStrategy: Equatable {
 	case Latest
 }
 
+extension SignalType where Value: SignalType, Error == Value.Error {
+	/// Flattens the inner signals sent upon `signal` (into a single signal of
+	/// values), according to the semantics of the given strategy.
+	///
+	/// If `signal` or an active inner signal emits an error, the returned
+	/// signal will forward that error immediately.
+	///
+	/// `Interrupted` events on inner signals will be treated like `Completed`
+	/// events on inner signals.
+	@warn_unused_result(message="Did you forget to call `observe` on the signal?")
+	public func flatten(strategy: FlattenStrategy) -> Signal<Value.Value, Error> {
+		switch strategy {
+		case .Merge:
+			return self.map(SignalProducer.init).merge()
+
+		case .Concat:
+			return self.map(SignalProducer.init).concat()
+
+		case .Latest:
+			return self.map(SignalProducer.init).switchToLatest()
+		}
```

Comment:
Since `flatten` is already defined for `Signal<SignalProducer<T, E>, E>`, you can just call `flatten` directly here.

```swift
return self.map(SignalProducer.init).flatten(strategy)
```